### PR TITLE
API: Deprecate OO plans.

### LINKED ISSUE #737: Un-document / deprecate old OO plans?
Before we had generators like `count` and `scan`, we had object-oriented implementations of plan logic, `Count` and `Scan`. In the spirit of self-consistency and paring down what there is to learn, it might be time to remove these. They add stateful list of subscriptions, flyers, and monitors, which are now better served by the preprocessors.

An OO-based plan is still sometimes a useful thing, so I think we should continue to ship some version of the base class `Plan`. But I don't think we need to ship ready-made OO counterparts for every plan. It's just one more thing to keep in sync.

Thoughts?

### PR DESCRIPTION
**Pull request by danielballan**
Closes #737

### CODE REVIEW

**Comment by danielballan** [comment]
@teddyrendahl @hhslepicka Do you know if anyone at SLAC using these OO plans? We prefer the generator-based ones, and we haven't been teaching people use the OO plans for well over a year now, but it's possible people discovered them on their own.

We have some simpler ideas about how to make "reusable" plans as opposed to single-use generator instances. See #731.

The OO plans are still used in some corners of NSLS-II (in old "acceptance tests" scripts at the beamlines) so we won't fully remove them for another release or two. This PR adds deprecation warnings and un-documents them. Nonetheless I wanted to check with you before we pull the trigger here.

I'm 99% sure our that APS collaborators haven't these and won't miss them, but let me know if I'm wrong, @prjemian.

**Comment by prjemian** [comment]
APS will not be bothered if the OO plans are dropped

On Aug 12, 2017 9:35 AM, "Dan Allan" <notifications@github.com> wrote:

@teddyrendahl <https://github.com/teddyrendahl> @hhslepicka
<https://github.com/hhslepicka> Do you know if anyone at SLAC using these
OO plans? We prefer the generator-based ones, and we haven't been teaching
people use the OO plans for well over a year now, but it's possible people
discovered them on their own.

We have some simpler ideas about how to make "reusable" plans as opposed to
single-use generator instances. See #731
<https://github.com/NSLS-II/bluesky/pull/731>.

The OO plans are still used in some corners of NSLS-II (in old "acceptance
tests" scripts at the beamlines) so we won't fully remove them for another
release or two. This PR adds deprecation warnings and un-documents them.
Nonetheless I wanted to check with you before we pull the trigger here.

I'm 99% sure our that APS collaborators haven't these and won't miss them,
but let me know if I'm wrong, @prjemian <https://github.com/prjemian>.

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/NSLS-II/bluesky/pull/743#issuecomment-321985167>, or mute
the thread

File: doc/source/api_changes.rst [hunk 1]
```diff
@@ -195,6 +195,8 @@ API Changes
 * The module ``bluesky.qt_kicker`` has been removed. Its former contents are
   avaiable in ``bluesky.utils``. The module was originally deprecated in April
   2016, and it has been issuing warnings about this change since.
+* The Object-Orenited plans (``Count``, ``Scan``, etc.) have been deprecated
```
Comment (stuartcampbell) [inline doc/source/api_changes.rst]: Minor typo (orientated)

**Comment by teddyrendahl** [comment]
@danielballan No use of OO plans at SLAC as far as my team is concerned.

**Comment by danielballan** [comment]
Cool. Thanks for the fast responses, @teddyrendahl and @prjemian.

**Pull request merged**